internal/mcpserver/agent/tool/agentrun_cache.go

```diff
@@ -0,0 +1,56 @@
+package tool
+
+import (
+	"sync"
+
+	"github.com/pluralsh/console/go/client"
+)
+
+// AgentRunCache provides thread-safe caching of AgentRunBaseFragment by runID
+type AgentRunCache struct {
+	mu   sync.RWMutex
+	data map[string]*client.AgentRunFragment
```

Comment:
Why does this need to be a map? There should only be one agent run at play throughout the lifetime of the entire process (also means this could probably just be a global var)